File: 2264. Largest 3-Same-Digit Number in String.java
class Solution {
    public String largestGoodInteger(String num) {
        String ans = "";
        int count = 1;
        char prev = num.charAt(0);
        for(int i = 1;i<num.length();i++){
            if(num.charAt(i) == prev){
                count++;
                if(count == 3){
                    String tmp = num.substring(i-2,i+1);
                    if(ans.compareTo(tmp) < 0) ans = tmp;
                }
            } else {
                count = 1;
                prev = num.charAt(i);
            }
        }
        return ans;
    }
}

File: 2264. Largest 3-Same-Digit Number in String.cpp
/*
Problem credits: https://leetcode.com/problems/largest-3-same-digit-number-in-string/description/

You are given a string num representing a large integer. An integer is good if it meets the following conditions:

It is a substring of num with length 3.
It consists of only one unique digit.
Return the maximum good integer as a string or an empty string "" if no such integer exists.

Note:

A substring is a contiguous sequence of characters within a string.
There may be leading zeroes in num or a good integer.
 

Solution:
    keep incremnting on seeing same char on prev index and get max number
*/
class Solution {
public:
    string largestGoodInteger(string num) {
        char ans = ' ';
        char prev = 'a';
        int cnt = 0;
        for(char c: num) {
            if (c == prev) {
                cnt++;
                if (cnt == 3) {
                    if (ans < c) {
                        ans = c;
                    }
                }
            } else {
                cnt = 1;
            }
            prev = c;
        }
        return (ans == ' ' ? "" : string(3, ans));
    }
};